the-collab-lab/tcl-69-smart-shopping-list
# [JH] Refactor list items from List to ManageList

## Description

This code separates all list items related logic and UI from List.jsx into ManageList.jsx. In addition, a value attribute was added to the share list email input field to reflect `recipientEmail` useState.  

Note: The adding of the 'Manage List' link in navbar will be handled in a separate issue. Please navigate to the page using '/manage-list' in the URL in the meantime. 

## Related Issue

Sub-issue of #14 

## Acceptance Criteria

- [x] List items are no longer visible in List.jsx
- [x] List items, including add items and share list features, are moved to a separate component

## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

## Updates

### Before
/list (List.jsx) 
<img width="1272" alt="Screen Shot 2024-04-01 at 5 19 20 PM" src="https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/64502078/f6b27efe-907d-4380-b92b-f3b0eaa87ece">

### After
/manage-list (ManageList.jsx)
<img width="1271" alt="Screen Shot 2024-04-01 at 5 19 36 PM" src="https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/64502078/f1fc66b5-86f4-4f14-b630-bbdcbee3ac87">
/list (List.jsx)
<img width="1264" alt="Screen Shot 2024-04-02 at 8 54 06 AM" src="https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/64502078/8f10787f-4dfc-43e5-970f-a72cfef392a0">

## Testing Steps / QA Criteria

1. Sign in.
2. Navigate to List page, the list items section should no longer be there.
3. Select a list.
4. Navigate to '/manage-list' in the url. List items should appear and function the same prior to refactoring.
